File: good-learner-backend/src/main/java/tech/nocountry/goodlearnerbackend/model/Teacher.java
package tech.nocountry.goodlearnerbackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "teacher")
@PrimaryKeyJoinColumn(name="person_id")
public class Teacher extends Person implements Serializable {

    @Column(name = "title")
    private String title;

    @Column(name = "school_entry_date")
    private LocalDate schoolEntryDate;

    public Teacher(
            String firstName,
            String lastName,
            String document,
            LocalDate birthDate,
            String email,
            LocalDateTime timeStamp,
            String phone,
            String title,
            LocalDate schoolEntryDate
    )
    {
        super(firstName, lastName, document, birthDate, email, timeStamp, phone);
        this.title = title;
        this.schoolEntryDate = schoolEntryDate;
    }

}

File: client/src/models/Teacher.ts
export class Teacher {
    idPerson?: number;
    firstName?: string;
    lastName?: string;
    document?: string;
    birthDate?: string;
    email?: string;
    timeStamp?: string;
    phone?: string;

    static parseItem(raw: any): Teacher {
        const teacher = new Teacher();
            teacher.idPerson = raw.idPerson ? raw.idPerson : undefined;
            teacher.firstName = raw.firstName ? raw.firstName : undefined;
            teacher.lastName = raw.lastName ? raw.lastName : undefined;
            teacher.document = raw.document ? raw.document : undefined;
            teacher.birthDate = raw.birthDate ? raw.birthDate : undefined;
            teacher.email = raw.email ? raw.email : undefined;
            teacher.timeStamp = raw.timeStamp ? raw.timeStamp : undefined;
            teacher.phone = raw.phone ? raw.phone : undefined;
        return teacher;
    }
}